recipes/unicorn/all/conanfile.py

```diff
@@ -0,0 +1,158 @@
+        if unsupported_archs:
+            self.output.info(f"Valid supported architectures are: {self._all_supported_archs}")
+            raise ConanInvalidConfiguration(f"Invalid arch(s) in supported_archs option: {unsupported_archs}")
+        if "arm" in self.settings.arch:
+            # FIXME: will/should be fixed with unicorn 2 (https://github.com/unicorn-engine/unicorn/issues/1379)
+            raise ConanInvalidConfiguration("arm builds of unicorn are currently unsupported")
+
+    def package_id(self):
+        # normalize the supported_archs option (sorted+comma separated)
+        self.info.options.supported_archs = ",".join(self._supported_archs)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    @property
+    def _jwasm_wrapper(self):
+        return os.path.join(self.build_folder, "jwasm_wrapper.py")
+
+    @functools.lru_cache(1)
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["UNICORN_INSTALL"] = True
+        cmake.definitions["UNICORN_BUILD_SAMPLES"] = False
+        cmake.definitions["UNICORN_ARCH"] = " ".join(self._supported_archs)
+        if self._needs_jwasm:
+            cmake.definitions["CMAKE_ASM_MASM_COMPILER"] = self._jwasm_wrapper
+            if self.settings.arch == "x86_64":
+                cmake.definitions["CMAKE_ASM_MASM_FLAGS"] = {
+                    "x86_64": "-win64",
+                    "x86": "-coff",
+                }[str(self.settings.arch)]
+        cmake.configure()
+        return cmake
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        if self._needs_jwasm:
+            tools.save(self._jwasm_wrapper, textwrap.dedent("""\
+                #!/usr/bin/env python
+                import os
+                import sys
+                import subprocess
+                args = ["jwasm"]
+                # Convert '/' options to '-'
+                for arg in sys.argv[1:]:
+                    if arg in ("-fpic", "-fPIC", "-Wall", "-fvisibility=hidden"):
+                        continue
+                    if os.path.exists(arg):
+                        args.append(arg)
+                    elif arg[:1] == "/":
+                        args.append("-"+arg[1:])
+                    else:
+                        args.append(arg)
+                print("args:", args)
+                subprocess.run(args, check=True)
+            """))
+            os.chmod(self._jwasm_wrapper, stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR | stat.S_IRGRP | stat.S_IWGRP | stat.S_IXGRP | stat.S_IROTH | stat.S_IXOTH)
+
+    def build(self):
+        self._patch_sources()
+        try:
+            cmake = self._configure_cmake()
+            cmake.build()
+        except:
```

Comment:
Let's capture the exception and show it, it can happen that it is raised from Conan before running the underlying CMake.

```suggestion
        except Exception as e:
            self.output.error("Exception while building: {}".format(e))
```

Reply:
Could we raise ConanException?